metaflow/plugins/argo/argo_workflows_cli.py

```diff
@@ -290,8 +299,41 @@ def create(
             obj.echo(
                 "Note that the flow was deployed with a modified name "
                 "due to Kubernetes naming conventions\non Argo Workflows. The "
-                "original flow name is stored in the workflow annotation.\n"
+                "original flow name is stored in the workflow annotations.\n"
+            )
+
+        if obj.workflow_name != obj._v1_workflow_name:
+            # Delete the old workflow if it exists
+            try:
+                ArgoWorkflows.delete(obj._v1_workflow_name)
+                obj.echo("Important!", bold=True, nl=False)
+                obj.echo(
+                    " To comply with new naming restrictions on Argo "
+                    "Workflows, this deployment replaced the\npreviously "
+                    "deployed workflow {v1_workflow_name}.\n".format(
+                        v1_workflow_name=obj._v1_workflow_name
```

Comment:
we could also check whether `v1_workflow_name` exists or not, and cut this alert out if  we did not replace an old deployment